### ISSUE TITLE
Add a single flag which says if exec flag should be set

### ISSUE THREAD

**Issue by tyler-gilbert** [StratifyLabs/StratifyOS#151]
```

//Tasks must meet the follow 4 criteria to be currently executing
        if ( task_enabled(i) && //Task is currently valid and enabled
             task_active_asserted(i) && //not blocked on anything
             !task_stopped_asserted(i) && //not stopped by a signal
             ( task_get_priority(i) == task_get_current_priority() ) ){ //task is equal to the currently executing priority
            //Enable process execution for highest active priority tasks
            task_assert_exec(i);
        } else {
            //Disable process execution for lower priority tasks
            task_deassert_exec(i);
        }
```

enabled(), active(), and not stopped() should update another flag so that only one flag needs to be checked.  If the task is disabled, the flag will be cleared. Then if the task is enabled (and the other flags are set accordingly), the flag will be set again.
